### PR TITLE
Add the plumbing in the agent forwarder to submit container images and SBOM

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

This PR adds the plumbing in the sender of the agent to be able to submit container images metadata and SBOM to their dedicated intakes.

* The container images message
  * is described in DataDog/agent-payload#196.
  * It will be sent to the URL defined in the `container_image.dd_url` parameter.
  * `container_image.additional_endpoints` can be used for dual shipping.
* The SBOM message
  * is described in DataDog/agent-payload#212.
  * It will be sent to the URL defined in the `sbom.dd_url` parameter.
  * `sbom.additional_endpoints` can be used for dual shipping.

### Motivation

This will be used by dedicated checks.
* Long-running core check to submit the container image metadata: #14567
* Long-running core check to submit the SBOM: #14989

### Additional Notes

This PR contains nothing original. It basically replicates what’s already done to submit data to other EventPlatform tracks and more particularly the `Container Lifecycle` one because the usage of those 3 tracks (the existing `contlcyle` track as well as the `contimage` and `sbom` ones implemented in this PR) are very similar.

### Possible Drawbacks / Trade-offs

The code base probably deserves a refactoring to make all the components involved in sending messages to Event Platform more generic in order to avoid having to duplicate so much plumbing code.
Such a refactoring requires some discussions and choices (Use of GO generics vs. `interface{}`, list of options to pass forwarder constructor, etc.) that might take time and that are not compatible with the timeline with the projects depending on this PR. That’s why I choose to do the refactoring after this PR rather than before.

### Describe how to test/QA your changes

Testing this change requires to also have the check that leverages it to submit payloads to the intake. This is in #14567.
Then, the [`[EventPlatform] Processing overview` dashboard](https://ddstaging.datadoghq.com/dashboard/zgu-732-67s/eventplatform-processing-overview?tpl_var_track_type%5B0%5D=contimage&from_ts=1671717321000&to_ts=1672926921000&live=true) can be used to validate that the payloads sent by the agent are properly received by the event platform.
The proper track type has to be selected in the template variable panel at the top. (`track_type` = `contimage` or `sbom`)

image

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by clamoriniere — APPROVED** [review]

**Review by songy23 — APPROVED** [review]

**Review by dinooliva — APPROVED** [review]

File: pkg/aggregator/demultiplexer_agent.go [hunk 1]
```diff
@@ -309,6 +331,24 @@ func (d *AgentDemultiplexer) Run() {
 			log.Debug("not starting the container lifecycle forwarder")
 		}
 
+		// container image forwarder
+		if d.forwarders.containerImage != nil {
+			if err := d.forwarders.containerImage.Start(); err != nil {
+				log.Errorf("error starting container image forwarder: %v", err)
+			}
+		} else {
+			log.Debug("no starting the container image forwarder")
```
Comment (Kaderinho) [inline pkg/aggregator/demultiplexer_agent.go:340]:
```suggestion
			log.Debug("not starting the container image forwarder")
```

File: pkg/aggregator/demultiplexer_agent.go [hunk 2]
```diff
@@ -309,6 +331,24 @@ func (d *AgentDemultiplexer) Run() {
...
+		}
+
+		// sbom forwarder
+		if d.forwarders.sbom != nil {
+			if err := d.forwarders.sbom.Start(); err != nil {
+				log.Errorf("error starting SBOM forwarder: %v", err)
+			}
+		} else {
+			log.Debug("no starting the SBOM forwarder")
```
Comment (Kaderinho) [inline pkg/aggregator/demultiplexer_agent.go:349]:
```suggestion
			log.Debug("not starting the SBOM forwarder")
```

File: pkg/aggregator/mocksender/asserts.go [hunk 3]
```diff
@@ -70,6 +72,16 @@ func (m *MockSender) AssertEventPlatformEvent(t *testing.T, expectedRawEvent str
 	return m.Mock.AssertCalled(t, "EventPlatformEvent", expectedRawEvent, expectedEventType)
 }
 
+// AssertContainerImage assert the expected event was emitted with the following values
+func (m *MockSender) AssertContainerImage(t *testing.T, expectedContainerImages []contimage.ContainerImagePayload) bool {
+	return m.Mock.AssertCalled(t, "ContainerImage", expectedContainerImages)
+}
+
+// AssertContainerImage assert the expected event was emitted with the following values
```
Comment (Kaderinho) [inline pkg/aggregator/mocksender/asserts.go:80]:
```suggestion
// AssertSBOM assert the expected event was emitted with the following values
```

File: pkg/aggregator/mocksender/mocked_methods.go [hunk 4]
```diff
@@ -117,6 +117,16 @@ func (m *MockSender) ContainerLifecycleEvent(msgs []serializer.ContainerLifecycl
 	m.Called(msgs)
 }
 
+// ContainerImage submit container image messages
+func (m *MockSender) ContainerImage(msgs []serializer.ContainerImageMessage) {
+	m.Called(msgs)
+}
+
+// SBOM submit container image messages
```
Comment (Kaderinho) [inline pkg/aggregator/mocksender/mocked_methods.go:125]: Not sure what to put for this one but I believe it's a copy paste ?

```suggestion
// SBOM submit sbom data
```

File: pkg/containerimage/forwarder.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+	if !conf.IsSet("container_image.additional_endpoints") {
+		return keysPerDomain, nil
+	}
+
+	additionalEndpoints := conf.GetStringMapStringSlice("container_image.additional_endpoints")
+
+	return config.MergeAdditionalEndpoints(keysPerDomain, additionalEndpoints)
+}
+
+// NewForwarder returns a forwarder for container lifecycle events
```
Comment (Kaderinho) [inline pkg/containerimage/forwarder.go:40]:
```suggestion
// NewForwarder returns a forwarder for container images events
```

File: pkg/forwarder/test_common.go [hunk 6]
```diff
@@ -183,3 +183,13 @@ func (tf *MockedForwarder) SubmitOrchestratorManifests(payload transaction.Bytes
 func (tf *MockedForwarder) SubmitContainerLifecycleEvents(payload transaction.BytesPayloads, extra http.Header) error {
 	return tf.Called(payload, extra).Error(0)
 }
+
+// SubmitContainerImages mock
+func (tf *MockedForwarder) SubmitContainerImages(payload transaction.BytesPayloads, extra http.Header) error {
+	return tf.Called(payload, extra).Error(0)
+}
+
+// SubmitSBOM
```
Comment (Kaderinho) [inline pkg/forwarder/test_common.go:192]:
```suggestion
// SubmitSBOM mock
```

**Review by Kaderinho — APPROVED** [review]
Small nits but LGTM

**Review by vickenty — APPROVED** [review]

**Pull request merged**